hadoop-hdds/container-service/src/main/proto/StorageContainerDatanodeProtocol.proto

```diff
@@ -239,6 +239,7 @@ message ContainerAction {
 
 message PipelineReport {
   required PipelineID pipelineID = 1;
+  optional bytes leaderID = 2;
```

Comment:
Can we also change the Pipeline in the proto file to bytes as well, so that we do not have to treat it as a string ?

Reply:
make sense why is it marked as required? Should we also mark it optional?

Reply:
Agree, but problem is on hdds side is google protobuf type and Ratis shades, so, if we make it ByteString in pipeline then on client side we will have to do RaftPeerId.valueOf(...), and make google ByteString to shaed one.